src/Hedgehog/Seed.fs

```diff
@@ -144,6 +144,20 @@ module Seed =
             crashUnless (k >= 0I) "k >= 0I"
             lo + v % k, seedN
 
+    /// Generates a random double in the specified range.
+    let rec nextDouble (lo : double) (hi : double) (seed : Seed) : double * Seed =
+        if lo > hi then
+            nextDouble hi lo seed
+        else
+            let x, seed' =
+                seed |> nextBigInt
+                    (bigint Int32.MinValue)
+                    (bigint Int32.MaxValue)
+            let scaled_x =
+                  (0.5 * lo + 0.5 * hi) + ((0.5 * hi - 0.5 * lo) / (0.5 * 4294967296.0)) * float x
```

Comment:
This is `randomIvalDouble` ported from Haskell [System.Random](https://hackage.haskell.org/package/random-1.1/docs/src/System-Random.html), similar to the existing `Seed.nextBigInt` which is really `randomIvalInteger` ported from that same module.